src/components/StatsLeaderboard.vue

```diff
@@ -144,13 +147,16 @@ export default {
           const matchesThisWeek = playerMatches[weekNum] || [];
           const wins = matchesThisWeek.filter((match) => match.result === Result.WON);
           res.total += wins.length;
-          res[`${weekNum}`] = wins.length;
+          res[`${weekNum}`] = {
+            wins: wins.length,
+            matchesPlayed: matchesThisWeek.length,
+          }
```

Comment:
To avoid excess repetition, I added the number of matches played by a user in a given week to `playerWins` instead of creating a new variable. It feels like it fits here

Reply:
This makes sense in the control flow but it smells a bit; I think mostly because the existing setup is already a bit funky. I haven't fully baked my thoughts on the issue, but two points feel a bit weird:

1. It seems odd that the playerWins computed prop includes matches that you lost. This is mostly a naming issue--maybe the computed prop could be called playerMatches and its constituent objects could have `winCount` and `matchCount` attributes

2. It feels a bit funky that we have match count, wins, and points as these relevant variables for each player and two out of three are grouped together while one (points) is separated. I wonder if it makes sense to set things up so that points makes its way into the same object as the point and match counts. That might be weird in terms of control flow, given that we need the first two to calculate the third. So maybe this change doesn't make sense. What do y'all think @peterancourt and @seriouslysean ?

Reply:
I like `playerMatches` instead. Avoid the naming issue-- I think they belong together but agree naming was the issue. Re: organization issue, it feels like `wins` and `matchesPlayed` belong together, with the consideration we change to count variables like @itsalaidbacklife said.

Since you need those two the calculate the third, it should probably become a variable you calculate in vue as a computed prop. Does the server really care about it at all?

Alternatively, would it be better to just return `matches` and have it be a flatMap of wins losses that you can calculate in Vue? Not sure the server needs to know unless we're putting together a completely full payload. Otherwise we can call player matches and do something to filter the mapping for wins or losses, or count it for total matches.

Reply:
I keep looking at this and I think I'm starting to agree with @itsalaidbacklife more-- `winCount` and `matchCount` make sense. Can we just do something like this in here (with the assumption that server side processing is faster):
```
res[`${weekNum}`] = {
            wins: wins.length,
            losses: matchesThisWeek.length - wins.length,
            matches: matchesThisWeek.length,
};
```
What about stalemates? Do they count as a loss here?

Reply:
There are no stalemate matches only incomplete matches (stalemate games don't count towards completing matches and a match doesn't end until one player gets two wins)

Reply:
There are no stalemate matches. A match can be incomplete in the database, but the endpoint serving the stats data filters them out, so all the matches here should be complete, and your losses are all the matches you didn't win. @peterancourt let's rename the computed prop `playerMatches` -- I could go either way on whether to include `losses` explicitly.

For explicitness sake, how about we call the attributes `winCount`, `lossCount`, `totalCount`?

Reply:
sounds good to me 👍